syz-manager/rpc.go

```diff
@@ -276,14 +341,21 @@ func (serv *RPCServer) updateFilteredCover(pcs []uint32) error {
 	return nil
 }
 
-func (serv *RPCServer) shutdownInstance(name string, crashed bool) []byte {
-	var runner *Runner
-	if val, _ := serv.runners.LoadAndDelete(name); val != nil {
-		runner = val.(*Runner)
-	} else {
-		return nil
+func (serv *RPCServer) createInstance(name string, injectLog chan<- []byte) {
+	runner := &Runner{
+		name:       name,
+		requests:   make(map[int64]Request),
+		injectLog:  injectLog,
+		injectStop: make(chan bool),
+	}
+	if _, loaded := serv.runners.LoadOrStore(name, runner); loaded {
+		panic(fmt.Sprintf("duplicate instance %s", name))
 	}
+}
 
+func (serv *RPCServer) shutdownInstance(name string, crashed bool) []byte {
+	runnerPtr, _ := serv.runners.LoadAndDelete(name)
+	runner := runnerPtr.(*Runner)
 	runner.mu.Lock()
```

Comment:
@dvyukov If I'm not wrong, now it will crash if there was no Runner with the given name. But is it not theoretically possible that we'd call shutdownIstance for a VM that did not (yet) call fuzzer connect?

Reply:
We now create the runner in createInstance method. So things are now stricter and easier to reason about.